examples/hello_rust/hello_rust_main.rs

```diff
@@ -21,54 +21,115 @@
+ 
+ /****************************************************************************
+  * Public functions
+  ****************************************************************************/
+ 
+ /****************************************************************************
+  * hello_rust_main
+  ****************************************************************************/
+ 
+ #[no_mangle]
+ pub extern "C" fn main(_argc: i32, _argv: *const *const u8) -> i32 {
+     
+     unsafe {
+         printf(b"Hello, Rust!!\n\0" as *const u8);
+ 
+         // Open the LED Driver
+         printf(b"Opening /dev/userleds\n\0" as *const u8);
+     
+         let fd = open(b"/dev/userleds\0" as *const u8, O_WRONLY);
+ 
+         if fd < 0 {
+             let errcode = 0;
+             printf(b"ERROR: Failed to open /dev/userleds: %d\n\0" as *const u8, errcode);
+             return 1; // EXIT_FAILURE from stdlib.h
+         }
 
-#![no_main]
-#![no_std]
+         puts(b"Set LED 0 to 1\0".as_ptr() as *const i8);
+         let ret = ioctl(fd, ULEDIOC_SETALL.into(), 1); // Convert ULEDIOC_SETALL (u16) to i32
 
-/****************************************************************************
- * Uses
- ****************************************************************************/
-
-use core::panic::PanicInfo;
-
-/****************************************************************************
- * Externs
- ****************************************************************************/
-
-extern "C" {
-    pub fn printf(format: *const u8, ...) -> i32;
-}
-
-/****************************************************************************
- * Private functions
- ****************************************************************************/
-
-/****************************************************************************
- * Panic handler (needed for [no_std] compilation)
- ****************************************************************************/
-
-#[panic_handler]
-fn panic(_panic: &PanicInfo<'_>) -> ! {
-    loop {}
-}
-
-/****************************************************************************
- * Public functions
- ****************************************************************************/
-
-/****************************************************************************
- * hello_rust_main
- ****************************************************************************/
-
-#[no_mangle]
-pub extern "C" fn hello_rust_main(_argc: i32, _argv: *const *const u8) -> i32 {
-    /* "Hello, Rust!!" using printf() from libc */
-
-    unsafe {
-        printf(b"Hello, Rust!!\n\0" as *const u8);
-    }
+         if ret < 0 {
+             let errcode = 0;
+             printf(b"ERROR: ioctl(ULEDIOC_SUPPORTED) failed: %d\n\0" as *const u8, errcode);
+             return 1; // EXIT_FAILURE from stdlib.h
+         }
+ 
+         // Sleep a while...
+         puts(b"Waiting...\0".as_ptr() as *const i8);
+         // In no_std, we can't use Rust's usleep() function, so we use unsafe instead
```

Comment:
```suggestion
         // In no_std, we can't use Rust's usleep() function, so we use unsafe instead
```

Remove the comment and blank line